src/model/transaction/addEntityToContentState.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule addEntityToContentState
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+const generateRandomKey = require('generateRandomKey');
+
+import type ContentState from 'ContentState';
+import type DraftEntityInstance from 'DraftEntityInstance';
+
+function addEntityToContentState(
+  contentState: ContentState,
+  instance: DraftEntityInstance
+): ContentState {
+  const newKey = generateRandomKey();
```

Comment:
At the moment that doesn't work with the way we re-create a `ContentState` instance from raw state. There an entity map is created manually which is then passed to `ContentState.createFromBlockArray`. Since we have no `contentState` yet we cannot utilize `contentState.addEntity`. How about we do not pass the entity map to `ContentState.createFromBlockArray`, but create the content state and then reduce the raw entity map by calling `contentState.addEntity`? Then we could definitely use a numerical counter.